src/common/url-parser.ts

```diff
@@ -10,6 +10,26 @@ export class UrlParser {
     public areURLHostNamesEqual(urlA: string, urlB: string): boolean {
         const urlAObj = new URL(urlA);
         const urlBObj = new URL(urlB);
-        return urlAObj.hostname === urlBObj.hostname;
+
+        if (this.areBothFileUrls(urlAObj, urlBObj)) {
+            return this.areFileUrlsEqual(urlA, urlB);
+        }
+
+        return this.areUrlProtocolsEqual(urlAObj, urlBObj) && urlAObj.hostname === urlBObj.hostname;
+    }
+
+    public areFileUrlsEqual(urlA: string, urlB: string): boolean {
```

Comment:
Will `areFileUrlsEqual` need to be public? If not, can we refactor so we don't create a `URL` twice for each string?

Reply:
The idea was to give `URLParser` a function that can be used in the future to support this kind of functionality. But for the issue that I am trying to fix here; it doesn't need to be a `public` function; so will make the changes that you suggested. 👍